svgdigitizer/electrochemistry/electrolyte.py

```diff
@@ -0,0 +1,197 @@
+# ********************************************************************
+from functools import cache
+from pHcalc.pHcalc import Acid, Neutral, System
+from astropy import units as u
+from svgdigitizer.electrochemistry.normalization import normalize_unit
+
+import pubchempy as pcp
+import requests
+import xml.etree.ElementTree as ET
+import re
+
+
+acid_base_dict = {
+    'VLTRZXGMWDSKGL-UHFFFAOYSA-N': {'name': 'HClO4', 'components': [Acid(pKa=[-1.6], charge=0)]},  # CRC handbook of chemistry and physics 2017
+    'QAOWNCQODCNURD-UHFFFAOYSA-N': {'name': 'H2SO4', 'components': [Acid(pKa=[-3, 1.9], charge=0)]},  # http://www.chem.wisc.edu/areas/reich/pkatable/pKa_compilation-1-Williams.pdf
+    'NBIIXXVUZAFLBC-UHFFFAOYSA-N': {'name': 'H3PO4', 'components': [Acid(pKa=[2.16, 7.21, 12.32], charge=0)]},  # CRC handbook of chemistry and physics 2017
+    'BDAGIHXWWSANSR-UHFFFAOYSA-N': {'name': 'HCOOH', 'components': [Acid(pKa=[3.75], charge=0)]},  # CRC handbook of chemistry and physics 2017
+    'QTBSBXVTEAMEQO-UHFFFAOYSA-N': {'name': 'acetic acid', 'components': [Acid(pKa=[4.78], charge=0)]},
+    'VEXZGXHMUGYJMC-UHFFFAOYSA-N': {'name': 'HCl', 'components': [Acid(pKa=[-6.2], charge=0)]},  # Analytical Chemistry, 43(7), 1971, S. 969–970.
+    'JUJWROOIHBZHMG-UHFFFAOYSA-N': {'name': 'pyridine', 'components': [Acid(pKa=[5.23], charge=1)]},  # CRC handbook of chemistry and physics 2017
+    'QGZKDVFQNNGYKY-UHFFFAOYSA-N': {'name': 'NH3', 'components': [Acid(pKa=[9.25], charge=1)]},  # CRC handbook of chemistry and physics 2017
+    'BVKZGUZCCUSVTD-UHFFFAOYSA-N': {'name': 'H2CO3', 'components': [Acid(pKa=[6.35, 10.33], charge=0)]},  # CRC handbook of chemistry and physics 2017
+    'HEMHJVSKTPXQMS-UHFFFAOYSA-M': {'name': 'NaOH', 'components': [Neutral(charge=1)]},
+    'FAPWRFPIFSIZLT-UHFFFAOYSA-M': {'name': 'NaCl', 'components': [Acid(pKa=[-6.2], charge=0), Neutral(charge=1)]},
+    'PMZURENOXWZQFD-UHFFFAOYSA-L': {'name': 'Na2SO4', 'components': [Acid(pKa=[-3, 1.9], charge=0), Neutral(charge=2)]},
+    'RYFMWSXOAZQYPI-UHFFFAOYSA-K': {'name': 'Na3PO4', 'components': [Acid(pKa=[2.16, 7.21, 12.32], charge=0), Neutral(charge=3)]},
+    'BNIILDVGGAEEIG-UHFFFAOYSA-L': {'name': 'Na2HPO4', 'components': [Acid(pKa=[2.16, 7.21, 12.32], charge=0), Neutral(charge=2)]},
+    'AJPJDKMHJJGVTQ-UHFFFAOYSA-M': {'name': 'NaH2PO4', 'components': [Acid(pKa=[2.16, 7.21, 12.32], charge=0), Neutral(charge=1)]},
+    }
+
+
+class Electrolyte():
+    r"""
+    Electrolyte
+    """
+    def __init__(self, electrolyte):
+        self.electrolyte = electrolyte
+
+    @property
+    @cache
+    def pH(self):
+        r"""
+        pH of the electrolyte.
+        """
+        try:
+            return self.electrolyte['pH']
+        except KeyError:
+            return self.estimate_pH()
+
+    def estimate_pH(self):
+        r"""
+        Estimates the pH of aquoues electrolytes.
+
+        EXAMPLES::
+
+            CID can be retrieved with InChIKey::
```

Comment:
Where is an InChi used in this example?

Reply:
What is an "InChIKey" actually? Should it not just be called "InChI".

Also, I think it would be very nice to either expand these abbreviations everywhere or link to some documentation, CID can just mean too many things in particular:
```suggestion
        EXAMPLES:

        The PubChem Compound ID (CID) can be retrieved with the International Chemical Identifier (InChI)::
```

But maybe that's just the opinion of someone who was no clue what this is all about ;)